### PR TITLE
Re-enabling builds of third-party containers

### PR DESCRIPTION
**Pull request by smlambert**
The builds of the dockerfiles was disabled temporarily.  This PR is a "work in progress" to re-enable them, so more open-source packages can be added.

### CODE REVIEW

File: thirdparty_containers/build.xml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+-->
+
+<project name="thirdparty_containers" default="build" basedir=".">
+	<taskdef resource="net/sf/antcontrib/antlib.xml" />
+	<description>
+		Build external tests
+	</description>
+
+	<!-- set properties for this build -->
+	<property name="DEST" value="${BUILD_ROOT}/performance" />
```
Comment (sophia-guo) [inline thirdparty_containers/build.xml]: Maybe you have noticed that value should be "${BUILD_ROOT}/thirdparty_containers"

**Comment by smlambert** [comment]
I updated my branch, the external tests "compile" but they still do not execute a target:
https://ci.adoptopenjdk.net/view/work%20in%20progress/job/test_personal/131/console

**Comment by sophia-guo** [comment]
Based on Shelley's branch, add test status, personal build:

https://ci.adoptopenjdk.net/view/work%20in%20progress/job/test_personal/135/console

23:34:06 ===============================================
23:34:06 Running test scala_test_SE80_0 ...
23:34:06 ===============================================
23:34:06 test with NoOptions
23:34:06 { docker run --rm -v /home/jenkins/workspace/test_personal/openjdkbinary/j2sdk-image/jre/bin/../..:/java  adoptopenjdk-scala-test:latest "jvm pos neg"; \
23:34:06 if [ $? -eq 0 ] ; then echo "scala_test_SE80_0""_PASSED"; else echo "scala_test_SE80_0""_FAILED"; fi; } 2>&1 | tee -a TestTargetResult
23:34:06 Unable to find image 'adoptopenjdk-scala-test:latest' locally
23:34:07 docker: Error response from daemon: pull access denied for adoptopenjdk-scala-test, repository does not exist or may require 'docker login'.
23:34:07 See 'docker run --help'.
23:34:07 scala_test_SE80_0_FAILED
23:34:07 
23:34:07 
23:34:07 TEST TARGETS SUMMARY
23:34:07 +++++++++++++++++++++++++++++++++++++++++++++++
23:34:07 TOTAL: 1   PASS: 0   FAIL: 1   SKIP: 0
23:34:07 
23:34:07 FAILED test targets:
23:34:07 	scala_test_SE80_0
23:34:07 +++++++++++++++++++++++++++++++++++++++++++++++

**Comment by TianyuZuo** [comment]
looking at the console output https://ci.adoptopenjdk.net/view/work%20in%20progress/job/test_personal/135/console

It seems the docker image was not successfully built 
```
23:34:05      [exec]  ---> Running in 1a8392d2c052
23:34:05      [exec] [91mcontainer_linux.go:265: starting container process caused "process_linux.go:368: container init caused \"process_linux.go:351: running prestart hook 0 caused \\\"fork/exec /usr/bin/dockerd (deleted): no such file or directory\\\"\""
23:34:05      [exec] oci runtime error: container_linux.go:265: starting container process caused "process_linux.go:368: container init caused \"process_linux.go:351: running prestart hook 0 caused \\\"fork/exec /usr/bin/dockerd (deleted): no such file or directory\\\"\""
```
re-running the whole scala build process on my machine and lab machine, both of them had no problems. Wondering which cause the image building process failure.


I didn't see incorrect cmds in the `build-image` target. Could you point it out?
```
	<target name="build_image" depends="clean_image" description="build scala test docker image">
		<exec executable="docker">
			<arg line="build -t adoptopenjdk-scala-test -f dockerfile/Dockerfile --pull ." />
		</exec>
	</target>
```


One minor question,  I initially didn't introduce the `thirdparty_containers/build.xml` file because I think building container tests' images should be a separate job. Building all docker images is going to exploded the whole build process and may crash the machine (depending on machine capabilities). 

If we add `thirdparty_containers/build.xml`, it will be automatically detected ant build if the `BUILD_LIST` is not specified. Are we going to define BUILD_LIST in every job in the future.

**Comment by TianyuZuo** [comment]
I suspect this is a know issue in Docker and need machine configuration to walk around this issue https://github.com/opencontainers/runc/issues/1343

**Comment by sophia-guo** [comment]
Sorry, I mean something wrong with "build_image target" job. The image cannot be found locally.

For template job we do a map in jenkinsfile, which will set BUILD_LIST according to target. That is it's transparent for test build job.

For personal build we have parameter to set BUILD_LIST.

**Comment by TianyuZuo** [comment]
Thanks for the info Sophia. This `thirdparty_containers/build.xml` is going to be fine as we will set BUILD_LIST separately (a.k.a according to targets).

File: thirdparty_containers/build.xml (hunk 1 above)
Comment (sophia-guo) [reply thirdparty_containers/build.xml]: Seems I have a good one with this changed.
https://ci.adoptopenjdk.net/view/work%20in%20progress/job/test_personal/137/console

**Comment by sophia-guo** [comment]
Docker Image fails on test-packet-x64-ubuntu-16.04-1, passed on test-scaleway-x64-ubuntu-16.04-1. Not sure the difference between those two machines. Remember recent system compiling failed on on one machine and passed on the other one. So are those machines set by playbooks from scratch? @gdams ?

**Comment by smlambert** [comment]
@sophia-guo - thanks for your updates/help, I have updated the PR and ready for another review.  As you note, the docker image builds, the scala tests run within it, there are 75 failing scala tests, which I will look into further in a separate issue/PR (Failed: Total 2496, Failed 75, Errors 0, Passed 2421)

**Pull request merged**